torch/distributed/_pin_memory_utils.py

```diff
@@ -0,0 +1,66 @@
+# mypy: allow-untyped-defs
+import logging
+import os
+from logging import getLogger
+
+import torch
+
+logger = getLogger()
+logger.setLevel(logging.INFO)
+
+
+# Allows retrying cudaHostRegister if it fails
+CKPT_PIN_ALLOW_RETRY = os.environ.get("CKPT_PIN_ALLOW_RETRY", "1") == "1"
+# Peeks last cudaError before pinning shared memory
+CKPT_PIN_PEEK_CUDA_ERROR = os.environ.get("CKPT_PIN_PEEK_CUDA_ERROR", "0") == "1"
+# Pops last cudaError before pinning shared memory
+CKPT_PIN_POP_CUDA_ERROR = os.environ.get("CKPT_PIN_POP_CUDA_ERROR", "0") == "1"
```

Comment:
Do we want to export these variables? I guess not. We can just move before where they are used. For example, `CKPT_PIN_ALLOW_RETRY` is not used at all.

Reply:
Also, should these be `TORCH_`-prefixed?